tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java

```diff
@@ -46,6 +47,54 @@ public CheckCLDR handleCheck(String path, String fullPath, String value, Options
                 }
             }
         }
+        // eg
+        //ldml/listPatterns/listPattern/listPatternPart[@type="start"]
+        //ldml/listPatterns/listPattern[@type="standard-short"]/listPatternPart[@type="2"]
+        if (path.startsWith("//ldml/listPatterns/listPattern") && !path.endsWith("/alias")) {
+            XPathParts parts = XPathParts.getFrozenInstance(path);
+            // check order, {0} must be before {1}
+
+            switch(parts.getAttributeValue(-1, "type")) {
+            case "start":
+                checkNothingAfter1(value, result);
+                break;
+            case "middle":
+                checkNothingBefore0(value, result);
+                checkNothingAfter1(value, result);
+                break;
+            case "end":
+                checkNothingBefore0(value, result);
+                break;
+            case "2":
+                int pos1 = value.indexOf("{0}");
+                int pos2 = value.indexOf("{1}");
+                if (pos1 > pos2) {
+                    result.add(new CheckStatus().setCause(this)
+                        .setMainType(CheckStatus.errorType)
+                        .setSubtype(Subtype.invalidPlaceHolder)
+                        .setMessage("Invalid list pattern «" + value + "»: the placeholder {0} must be before {1}."));
+                }
+            }
```

Comment:
The spec also allows a special "3" case, which currently doesn't exist but should be checked:

```suggestion
            case "3":
                int pos1 = value.indexOf("{0}");
                int pos2 = value.indexOf("{1}");
                int pos3 = value.indexOf("{2}");
                if (pos1 > pos2 || pos2 > pos3) {
                    result.add(new CheckStatus().setCause(this)
                        .setMainType(CheckStatus.errorType)
                        .setSubtype(Subtype.invalidPlaceHolder)
                        .setMessage("Invalid list pattern «" + value + "»: the placeholders {0}, {1}, {2} must appear in that order."));
                }
            }